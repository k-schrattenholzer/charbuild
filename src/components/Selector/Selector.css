.Selector-Container {
  display: grid;
  grid-template-columns: 1fr;
  background-color: var(--jet-2);
  border-radius: 30px;
  padding: 10px;
  gap: 5px;
  align-items: center;
  color: var(--jet);
  font-size: 1.5em;
}

.Options-Container {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  /* background-color: #706e53;
  border-radius: 30px; */
  padding: 5px;
  align-items:flex-end;
}

.Option {
  display: flex;
  flex-direction: column;
  width: 5em;
  background-color: var(--laurel-green-2);
  border-radius: 30px;
  padding: 10px;
  box-shadow: var(--russian-green) 0px 2px 10px;
  text-shadow: -1px 2px 0 var(--bone);
}