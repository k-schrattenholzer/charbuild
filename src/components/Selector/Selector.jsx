import React from 'react';
import './Selector.css'

export default function Selector({
  head,
  onHeadChange,
  mid,
  onMidChange,
  bottom,
  onBottomChange,
  shoes,
  onShoeChange,
  newSaying,
  addSaying,
  handleClick,
}) {

  const headIcons = ['bat', 'bearded dragon', 'frog', 'parrot', 'pigeon', 'snake', 'walrus'];
  const midIcons = ['dress', 'jacket', 'poncho', 'shirt', 'sweater'];
  const bottomIcons = ['jeans', 'pijamas', 'shorts'];
  const shoeIcons = ['boot', 'converse', 'sandals', 'tennies']

  return (
    <div 
    className='Selector-Container'
    >
      <article
      className='Options-Container'>
        <label
        className='Option'>
          face-space
          <select 
          value={head} 
          onChange={(e) => onHeadChange(e.target.value)}>
              {headIcons.map((png) => (
              <option key={png}>{png}</option>
            ))}
            </select>
        </label>
        <label
        className='Option'>
          tops
          <select 
          value={mid} 
          onChange={(e) => onMidChange(e.target.value)}>
              {midIcons.map((png) => (
              <option key={png}>{png}</option>
            ))}
            </select>
        </label>
        <label
        className='Option'>
          bottoms
          <select 
          value={bottom} 
          onChange={(e) => onBottomChange(e.target.value)}>
              {bottomIcons.map((png) => (
              <option key={png}>{png}</option>
            ))}
            </select>
        </label>
        <label
        className='Option'>
          shoes
          <select 
          value={shoes} 
          onChange={(e) => onShoeChange(e.target.value)}>
              {shoeIcons.map((png) => (
              <option key={png}>{png}</option>
            ))}
            </select>
        </label>
        <label
        className='Option'>
          whats on yr brain-space?
          <input
            type='text'
            value={newSaying}
            onChange={(e) => addSaying(e.target.value)}
            // onFocus={(e) => {
            //   this.value = '';
            // }}
          />
          <button 
            type='button'
            onClick={handleClick}>
                    put it in the bucket
          </button>
        </label>  
      </article>
    </div>
  )
}
