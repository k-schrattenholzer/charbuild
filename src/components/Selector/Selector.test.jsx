import { screen, render } from "@testing-library/react";
import Selector from "./Selector";

it('should render the selector component', () => {
  const {container} = render(<Selector />);
  const selectTops = screen.getByLabelText(/tops/);
  const selectBottoms = screen.getByLabelText(/bottoms/);
  const selectShoes = screen.getByLabelText(/shoes/);
  const input = screen.getByLabelText(/whats on yr brain-space?/);

  expect(selectTops).toBeInTheDocument();
  expect(selectBottoms).toBeInTheDocument();
  expect(selectShoes).toBeInTheDocument();
  expect(input).toBeInTheDocument();
  expect(container).toMatchSnapshot();
})