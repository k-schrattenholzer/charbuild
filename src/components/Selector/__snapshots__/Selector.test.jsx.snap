// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the selector component 1`] = `
<div>
  <div
    class="Selector-Container"
  >
    <h4>
      yr cutie
    </h4>
    <article
      class="Options-Container"
    >
      <label
        class="Option"
      >
        face-space
        <select>
          <option>
            bat
          </option>
          <option>
            bearded dragon
          </option>
          <option>
            frog
          </option>
          <option>
            parrot
          </option>
          <option>
            pigeon
          </option>
          <option>
            snake
          </option>
          <option>
            walrus
          </option>
        </select>
      </label>
      <label
        class="Option"
      >
        tops
        <select>
          <option>
            dress
          </option>
          <option>
            jacket
          </option>
          <option>
            poncho
          </option>
          <option>
            shirt
          </option>
          <option>
            sweater
          </option>
        </select>
      </label>
      <label
        class="Option"
      >
        bottoms
        <select>
          <option>
            jeans
          </option>
          <option>
            pijamas
          </option>
          <option>
            shorts
          </option>
        </select>
      </label>
      <label
        class="Option"
      >
        shoes
        <select>
          <option>
            boot
          </option>
          <option>
            converse
          </option>
          <option>
            sandals
          </option>
          <option>
            tennies
          </option>
        </select>
      </label>
      <label
        class="Option"
      >
        whats on yr brain-space?
        <input
          type="text"
          value=""
        />
      </label>
      <button
        type="button"
      >
        put it in the bucket
      </button>
    </article>
  </div>
</div>
`;
